src/apps/drag-and-drop/DragAndDropApp.tsx

```diff
@@ -0,0 +1,47 @@
+import {DragAndDropList} from "../../components/drag-and-drop-list/DragAndDropList";
+import "./drag-and-drop.css"
+import {useState} from "react";
+
+
+export function DragAndDropApp() {
+  const [listOneItems, setListOneItems] = useState<Array<string>>(['item 1', 'item 2', 'item 3', '--']);
+  const [listTwoItems, setListTwoItems] = useState<Array<string>>(['item 4', 'item 5', 'item 6', '--']);
+
+  function onItemDropped(item: string, sourceListId: number, targetListId: number) {
+    if (sourceListId === targetListId) return
+
+    if (sourceListId === 1 && targetListId === 2) {
+      setListOneItems(listOneItems.filter(i => i !== item));
```

Comment:
Please use the prev value inside the callback for this. This will give you the latest state.
```suggestion
      setListOneItems((prev) => prev.filter(i => i !== item));
```